src/groups/special_euclidean.jl

```diff
@@ -494,6 +549,128 @@ function exp_lie!(G::SpecialEuclidean{TypeParameter{Tuple{3}}}, q, X)
     return q
 end
 
+@doc raw"""
+    jacobian_exp_inv_argument(
+        M::SpecialEuclidean{TypeParameter{Tuple{2}}},
+        p,
+        X,
+    )
+
+Compute Jacobian matrix of the invariant exponential map on [`SpecialEuclidean`](@ref)`(2)`.
+The formula reads
+````math
+\begin{pmatrix}
+\frac{\sin(θ)}{\theta} & \frac{1-\cos(\theta)}{\theta} & \frac{t_1(\sin(θ) - θ) + t_2(\cos(θ) - 1)}{\sqrt{2} θ^2} \\
+\frac{-1+\cos(\theta)}{\theta} & \frac{\sin(θ)}{\theta} & \frac{t_2(\sin(θ) - θ) + t_1(-\cos(θ) + 1)}{\sqrt{2} θ^2} \\
+0 & 0 & 1
+\end{pmatrix}.
+````
+where ``θ`` is the norm of `X` and ``[t_1, t_2]`` is the translation part
+of `X`.
+It is adapted from [Chirikjian:2012](@cite), Section 10.6.2, to `Manifolds.jl` conventions.
+"""
+jacobian_exp_inv_argument(M::SpecialEuclidean{TypeParameter{Tuple{2}}}, p, X)
+@doc raw"""
+    jacobian_exp_inv_argument(
+        M::SpecialEuclidean{TypeParameter{Tuple{3}}},
+        p,
+        X,
+    )
+
+Compute Jacobian matrix of the invariant exponential map on [`SpecialEuclidean`](@ref)`(3)`.
+The formula reads
+````math
+\begin{pmatrix}
+R & Q \\
+0_{3×3} & R
+\end{pmatrix},
+````
+where ``R`` is the Jacobian of exponential map on [`Rotations`](@ref)`(3)` with respect to
+the argument, and ``Q`` is
+````math
+\frac{1}{2} T - \frac{θ - \sin(θ)}{θ^3} (X_r T + T X_r + X_r T X_r) + \frac{1 - \frac{θ^2}{2} - \cos(θ)}{θ^4} (X_r^2 T + T X_r^2 - 3 X_r T X_r) + \frac{1}{2}\left(\frac{1 - \frac{θ^2}{2} - \cos(θ)}{θ^4} - 3 \frac{θ - \sin(θ) - \frac{θ^3}{6}}{θ^5}\right) (X_r T X_r^2 + X_r^2 T X_r)
```

Comment:
```suggestion
Q = \frac{1}{2} T - \frac{θ - \sin(θ)}{θ^3} (X_r T + T X_r + X_r T X_r) + \frac{1 - \frac{θ^2}{2} - \cos(θ)}{θ^4} (X_r^2 T + T X_r^2 - 3 X_r T X_r) + \frac{1}{2}\left(\frac{1 - \frac{θ^2}{2} - \cos(θ)}{θ^4} - 3 \frac{θ - \sin(θ) - \frac{θ^3}{6}}{θ^5}\right) (X_r T X_r^2 + X_r^2 T X_r)
```

and maybe we cloud split this in a few lines? This is very long.

Reply:
Sure, I can do that.